packages/opentelemetry-exporter-collector-grpc/src/CollectorMetricExporter.ts

```diff
@@ -38,6 +39,15 @@ export class CollectorMetricExporter
   // Converts time to nanoseconds
   protected readonly _startTime = new Date().getTime() * 1000000;
 
+  constructor(config: CollectorExporterConfigNode = {}) {
+    super(config);
+    const headers = baggageUtils.parseKeyPairsIntoRecord(getEnv().OTEL_EXPORTER_OTLP_METRICS_HEADERS);
```

Comment:
it doesn't make sense to keep the same code in 2 places (trace & metric), instead you could do it once in `CollectorExporterNodeBase`